src/components/slide-toggle/slide-toggle.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  Component,
+  ElementRef,
+  Renderer,
+  forwardRef,
+  ChangeDetectionStrategy,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR
+} from '@angular/common';
+import { BooleanFieldValue } from '../../core/annotations/field-value';
+import { Observable } from 'rxjs/Observable';
+
+export const MD_SLIDE_TOGGLE_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => MdSlideToggle),
+  multi: true
+};
+
+// Increasing integer for generating unique ids for slide-toggle components.
+let nextId = 0;
+
+@Component({
+  selector: 'md-slide-toggle',
+  host: {
+    '[class.md-checked]': 'checked',
+    '[class.md-disabled]': 'disabled',
+  },
+  templateUrl: './components/slide-toggle/slide-toggle.html',
+  styleUrls: ['./components/slide-toggle/slide-toggle.css'],
+  providers: [MD_SLIDE_TOGGLE_VALUE_ACCESSOR],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MdSlideToggle implements ControlValueAccessor {
+
+  private onChange = (_: any) => {};
+  private onTouched = () => {};
+
+  // A unique id for the slide-toggle. By default the id is auto-generated.
+  private _uniqueId = `md-slide-toggle-${++nextId}`;

```

Comment:
You can just initialize `id` to this without the `_uniqueId` field; if the user gives their own id, it will overwrite this (which is fine).